### ISSUE TITLE
Improve Dataproc instructions

### ISSUE THREAD

**Issue by jacobtomlinson** [rapidsai/deployment#305]
Following the latest [Dataproc](https://docs.rapids.ai/deployment/nightly/cloud/gcp/dataproc/) docs works, but doesn't give the best experience. We could improve the docs in a few ways.

1. By default you get RAPIDS 22.10. You can override this with a metadata flag, we should document this.

```
...
--metadata gpu-driver-provider=NVIDIA,dask-runtime=$DASK_RUNTIME,rapids-runtime=DASK,rapids-version=23.12\
...
```

2. Once things are set up if you use Jupyter Lab RAPIDS is not installed in the base environment. Also the RAPIDS environment isn't registered as a kernel that you can select in the notebook.

3. We should show how to connect to the Dask cluster in a notebook and run some work.

**Comment by skirui-source** [comment]
UPDATES: 

1.  I was able to successfully launch a dataproc cluster with rapids v23.12 stable, cuda 11.8 (both parsed via the metadata flags `--rapids-version` and `--cuda-version`)

2. The [install_gpu_driver.sh](https://github.com/GoogleCloudDataproc/initialization-actions/blob/950ad89138667339a9ad27447d83d0069be17781/gpu/install_gpu_driver.sh#L43)  script downloads ubuntu 18.04, and has outdated versions of cuda and gpu drivers ; it installs cuda11.2 by default doesn't include cuda 12 as an option 

3. The latest rapids 24.02 is only compatible with ubuntu 20.04 and 22.04; Thus the install scripts need to be updated accordingly with the newer drivers as well


4. To test RAPIDS libraries in the notebook environment, we need to edit [rapids.sh](https://github.com/GoogleCloudDataproc/initialization-actions/blob/master/rapids/rapids.sh#L65 ) script to activate the conda environment (`dask-rapids`) and register it as a kernel in Jupyter

*****

I have also submitted  [issue](https://github.com/GoogleCloudDataproc/initialization-actions/issues/1137) to the Google team to track the upgrade  all these initialization scripts
